### PR TITLE
Pass status along ipam update

### PR DESCRIPTION
**Pull request by LionelJouin**
This PR fixes the missing parts of #1050 (pass STATUS to a delegated plugin)
I also updated the cni dependency to v1.2.3

### CODE REVIEW

**Review by squeed — APPROVED** [review]
nice!!!!

File: plugins/main/host-device/host-device.go [hunk 1]
```diff
@@ -628,3 +629,20 @@ func validateCniContainerInterface(intf current.Interface) error {
...
+		return fmt.Errorf("failed to load netconf: %w", err)
+	}
+
+	if conf.IPAM.Type != "" {
+		if err := ipam.ExecStatus(conf.IPAM.Type, args.StdinData); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Check if hode device exists.
```
[the remark this answers is not in the document]
Comment (LionelJouin) [reply plugins/main/host-device/host-device.go]: I fixed the miss-spelling.
The idea comes from this comment: https://github.com/containernetworking/plugins/pull/1021#discussion_r1546029716

**Comment by champtar** [comment]
@LionelJouin you need to be rebase

**Comment by squeed** [comment]
@LionelJouin needs another rebase :-)

**Comment by squeed** [comment]
@LionelJouin somehow this PR still does not have the green button. Perhaps the latest changes to `go.mod` are causing the issue.

Can you rebase and drop your merge commit? That helps keep the history clean.

**Review by squeed — APPROVED** [review]

**Comment by LionelJouin** [comment]
I saw the tests of the VRF plugin are also flaky, I haven't looked into them yet.

**Pull request merged**